src/Sarif.Driver/Sdk/ExtensionMethods.cs

```diff
@@ -108,14 +111,37 @@ public static string GetFirstSentence(string text)
                 {
                     case '\'':
                     {
-                        insideQuotes = !insideQuotes;
+                        // we'll ignore everything within parenthized text
+                        if (!withinParentheses)
+                        {
+                            withinQuotes = !withinQuotes;
+                        }
+                        break;
+                    }
+
+                    case '(':
+                    {
+                        if (!withinQuotes)
+                        {
+                            withinParentheses = true;
+                        }
                         break;
                     }
 
+                    case ')':
+                    {
+                        if (!withinQuotes)
+                        {
+                            withinParentheses = false;
+                        }
+                        break;
+                    }
+                    case '\n':
+                    case '\r':
                     case '.':
                     {
-                        if (insideQuotes) { continue; }
-                        return text.Substring(0, length);
+                        if (withinQuotes || withinParentheses) { continue; }
+                        return text.Substring(0, length).TrimEnd('\r', '\n');
```

Comment:
This code exists because \r or \n explicitly marks the end of a sentence  but should not be included. Similarly, '.' marks the end of a sentence but should be included. So, this code isn't intended to remove extraneous whitespace but to accurately reflect where we draw the 'line' for the sentence.